### PR TITLE
ROX-19048: Integrate Scanner V4 into secured-cluster-services Chart

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

This PR adds Scanner V4 support to the secured-cluster-services Helm chart.
Since secured clusters can run in "local scanner" mode, which depends on scanner certificates being issued by central, this PR also extends the certificate issuing code within central.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~~Evaluated and added CHANGELOG entry if required~~
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

Feature is not public yet.

### Here I tell how I validated my change

Manual testing:

1. Preparations:
a. Connect to cluster, e.g. Infra GKE with instance type `e2-standard-8`
b. Make sure `$QUAY_USERNAME` and `$QUAY_TOKEN` contain valid quay.io credentials.
c. Set `export MAIN_IMAGE_TAG=...`.
d. Prepare the charts:
```
$ make cli
$ roxctl helm output central-services --output-dir ./stackrox-central-services-chart --remove --debug
$ roxctl helm output secured-cluster-services --output-dir ./stackrox-secured-cluster-services-chart --remove --debug
```
2. Install central-services:
a. I use the following chart configuration stored as `values-scannerv4.yaml`:
```
central:
  resources:
    requests:
      cpu: 400m
      memory: 1Gi
    limits:
      cpu: 400m
      memory: 1Gi
  image:
    tag: "$MAIN_IMAGE_TAG"
  persistence:
    none: true
  db:
    resources:
      requests:
        cpu: 400m
        memory: 1Gi
      limits:
        cpu: 400m 
        memory: 1Gi
    image:
      tag: "$MAIN_IMAGE_TAG"
  
  telemetry:
    enabled: false 

scanner:
  disable: false #true
  
scannerV4:
  disable: false 
  indexer:
    image:
      tag: "$MAIN_IMAGE_TAG"
  matcher:
    image:
      tag: "$MAIN_IMAGE_TAG"
  db:
    image:
      tag: "$MAIN_IMAGE_TAG"
  
imagePullSecrets:
  username: "$QUAY_USER"
  password: "$QUAY_TOKEN"

allowNonstandardNamespace: true
```
b. Set namespace: `export NS=stackrox`
c. run helm install:
```
helm install -n "$NS" --create-namespace --set central.exposure.loadBalancer.enabled=true --values <(envsubst < values-scannerv4.yaml) stackrox-central-services ./stackrox-central-services-chart
```
d. Set password: `export ROX_ADMIN_PASSWORD=<as produced by helm install>`
e. Obtain central endpoint:
```
export CENTRAL_ENDPOINT="https://$(kubectl -n stackrox get service central-loadbalancer -o jsonpath='{.status.loadBalancer.ingress[0].ip}'):443"
```
3. Install secured-cluster-services
a. I use the following chart configuration stored as `values-scannerv4-sc.yaml`:
```
clusterName: sc0
centralEndpoint: "$CENTRAL_ENDPOINT" 

image:
  main:
    tag: "$MAIN_IMAGE_TAG"
  collector:
    tag: "$MAIN_IMAGE_TAG"
  scannerV4:
    tag: "$MAIN_IMAGE_TAG"
  scannerV4DB:
    tag: "$MAIN_IMAGE_TAG"

sensor:
  resources:
    requests:
      cpu: 500m
      memory: 1Gi
    limits:
      cpu: 1000m
      memory: 2Gi

scanner:
  disable: false

scannerV4:
  disable: false 

imagePullSecrets:
  username: "$QUAY_USER"
  password: "$QUAY_TOKEN"

allowNonstandardNamespace: true
```
b. Set namespace: `export NS=stackrox-sc`
c. retrieve init bundle:
```
roxctl -e "$CENTRAL_ENDPOINT" --insecure-skip-tls-verify central init-bundles generate sc0 --output init-bundle-sc0.yaml
```
d. run helm install:
```
helm install -n "$NS" --create-namespace --values <(envsubst < values-scannerv4-sc.yaml) --set centralEndpoint="$CENTRAL_ENDPOINT" -f init-bundle-sc0.yaml stackrox-secured-cluster-services ./stackrox-secured-cluster-services-chart
```

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl [hunk 1]
```diff
@@ -34,19 +34,22 @@
...
-[< if (not .KubectlOutput) ->]
+{{/* Only merge scanner config shapes if kubectl output is disabled */}}
+[<- if not .KubectlOutput >]
   {{ $configShapeScanner := $.Files.Get "internal/scanner-config-shape.yaml" | fromYaml }}
-  {{ $_ := include "srox.mergeInto" (list $rox $configShape $configShapeScanner) }}
-[< else >]
-  {{ $_ := include "srox.mergeInto" (list $rox $configShape) }}
-[< end >]
+  {{ $_ = include "srox.mergeInto" (list $configShape $configShapeScanner) }}
+  {{/* Only merge scanner-v4 config shapes if kubectl output is disabled and feature flag is enabled */}}
```
Comment (RTann) [inline image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl:44]: nit: `scannerV4` to match the config-shape?

File: image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl [hunk 2]
```diff
@@ -257,6 +263,47 @@
 {{ end }}
 [<- end >]
 
+[<- if and (not .KubectlOutput) .FeatureFlags.ROX_SCANNER_V4 >]
+{{/*
+   Local scanner-v4-indexer setup.
```
Comment (RTann) [inline image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl:268]: nit: Scanner v4 Indexer

**Review by RTann — COMMENTED** [review]
didn't see all files yet, but here is a preliminary review

File: central/localscanner/certificates.go [hunk 3]
```diff
@@ -26,16 +28,39 @@ func IssueLocalScannerCerts(namespace string, clusterID string) (*storage.TypedS
...
+	// TODO: Under what circumstances exactly do we issue certs for Scanner V4 here?
+	// * Central could have support for Scanner V4 enabled in one way or another.
+	// * Sensor could have support or Scanner V4 enabled in one way or another.
+	// * Sensor could inform Central when requesting localscanner certs for which services new certs are required.
+	if features.ScannerV4.Enabled() {
+		_, scannerV4IndexerCertificate, err := localScannerCertificatesFor(storage.ServiceType_SCANNER_V4_INDEXER_SERVICE, namespace, clusterID)
+		if err != nil {
+			certIssueError = multierror.Append(certIssueError, err)
+		}
+		_, scannerV4MatcherCertificate, err := localScannerCertificatesFor(storage.ServiceType_SCANNER_V4_MATCHER_SERVICE, namespace, clusterID)
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply central/localscanner/certificates.go:46]: right..

File: central/localscanner/certificates.go [hunk 4]
```diff
@@ -56,7 +81,19 @@ func localScannerCertificatesFor(serviceType storage.ServiceType, namespace stri
 }
 
 func generateServiceCertMap(serviceType storage.ServiceType, namespace string, clusterID string) (secretDataMap, error) {
-	if serviceType != storage.ServiceType_SCANNER_SERVICE && serviceType != storage.ServiceType_SCANNER_DB_SERVICE {
+	var err error
+	supportedServices := []storage.ServiceType{
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply central/localscanner/certificates.go:85]: done

File: central/localscanner/certificates.go [hunk 5]
```diff
@@ -56,7 +81,19 @@ func localScannerCertificatesFor(serviceType storage.ServiceType, namespace stri
...
+		storage.ServiceType_SCANNER_SERVICE,
+		storage.ServiceType_SCANNER_DB_SERVICE,
+	}
+	if features.ScannerV4.Enabled() {
+		supportedServices = append(supportedServices,
+			storage.ServiceType_SCANNER_V4_INDEXER_SERVICE,
+			storage.ServiceType_SCANNER_V4_MATCHER_SERVICE,
+			storage.ServiceType_SCANNER_V4_DB_SERVICE)
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply central/localscanner/certificates.go:93]: 👍

File: image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml [hunk 6]
```diff
@@ -141,6 +142,24 @@ customize:
     podLabels: {} # dict
     podAnnotations: {} # dict
     envVars: {} # dict
+  scanner-v4-indexer:
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml:145]: Well, the keys here are expected to match the resource names exactly in order for this customization machinery to just work. That's also the reason why in line 133 the key is `admission-control` and not `admissionControl`.

File: image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl (hunk 2 above)
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl:268]: changed

File: pkg/scanners/clairv4/clairv4.go [hunk 7]
```diff
@@ -70,6 +70,17 @@ func newScanner(integration *storage.ImageIntegration, activeRegistries registri
 	if err := validate(cfg); err != nil {
 		return nil, err
 	}
+	// TODO: Probably something along these lines is required here? // MC
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply pkg/scanners/clairv4/clairv4.go:73]: removed

File: proto/storage/service_identity.proto [hunk 8]
```diff
@@ -31,6 +31,9 @@ enum ServiceType {
     BENCHMARK_SERVICE         = 8;
     SCANNER_SERVICE           = 9;
     SCANNER_DB_SERVICE        = 10;
+    SCANNER_V4_INDEXER_SERVICE = 13;
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply proto/storage/service_identity.proto:34]: done

**Comment by mclasmeier** [comment]
Marked as ready so that images are built and pushed.

File: proto/api/v1/credential_expiry_service.proto [hunk 9]
```diff
@@ -12,9 +12,9 @@ package v1;
 
 message GetCertExpiry {
     enum Component {
-        UNKNOWN = 0;
-        CENTRAL = 1;
-        SCANNER = 2;
+        UNKNOWN    = 0;
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply proto/api/v1/credential_expiry_service.proto]: reverted :)

File: image/templates/helm/stackrox-secured-cluster/templates/00-storage-class.yaml
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/templates/00-storage-class.yaml:1]: I think it is the same? But anyway, I have removed this file from the PR and move the already-existing file into the `shared/templates` folder.

File: image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml:1]: The config shape for the `scannerV4` structure came from a [shared file](https://github.com/stackrox/stackrox/blob/4a4edd03329a8f9c0047865c580905f21f6f1531/image/templates/helm/shared/internal/scanner-v4-config-shape.yaml)
and its content is merged [here](https://github.com/stackrox/stackrox/blob/c67658ea5c0f65833fb1ec4eee7d274e6447616d/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L47).

File: sensor/kubernetes/localscanner/service_certificates_repository_test.go [hunk 10]
```diff
@@ -217,6 +219,51 @@ func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertsUnknownServiceT
 	s.Error(err)
 }
 
+func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertsMultipleScannerSecrets() {
+	if !features.ScannerV4.Enabled() {
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply sensor/kubernetes/localscanner/service_certificates_repository_test.go:223]: According to @dcaravel's suggestion this feature flag is defined as unchangeable.
Out of curiosity, wha does this have to do with (non-)release builds?

File: central/credentialexpiry/service/service.go [hunk 11]
```diff
@@ -35,8 +35,19 @@ func New(imageIntegrations imageIntegrationStore.DataStore) Service {
 		// scanner must have patched their deployment to not hit this.
 		// At the same time, we don't want to make this a fatal error, so just log a warning.
 		log.Warnf("Failed to initialize scanner TLS config: %v", err)
-		tlsConfig = nil
+		scannerTLSConfig = nil
+	}
+	scannerV4IndexerTLSConfig, err := clientconn.TLSConfig(mtls.ScannerV4IndexerSubject, clientconn.TLSConfigOptions{
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply central/credentialexpiry/service/service.go:40]: The credential expiry service is used by the UI to inform the user about certificates of ACS services which are soon to expire. As I see it Scanner V4 components should be added there. But first scanner V4 needs to be added as an imageIntegration.

Sidenote: I don't understand this 100% yet. For example, what about Scanner DB (v2)? It also has a certificate that can expire.

Should we add Scanner V4 Indexer, Matcher and DB?

File: image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl (hunk 1 above)
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl:44]: Are you only referring the comment here?

**Comment by mclasmeier** [comment]
I'd say CI is looking good.

File: image/templates/helm/stackrox-secured-cluster/values-scanner-v4-indexer.yaml.example [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+# image:
+#  # Configuration for the `scanner-v4` image that is used by Scanner v4 Indexer.
+#  scanner-v4:
+#    indexer:
+#      registry: null
+#      name: scanner-slim
+#      tag: null
+#      fullRef: null
+#
+# scanner-v4:
```
Comment (johannes94) [inline image/templates/helm/stackrox-secured-cluster/values-scanner-v4-indexer.yaml.example:23]: For central helm chart we were using scannerV4 in the values file, and I think even in this PR we have different namings, some using kebab case some using camel case, I see the explanation above on @RTann, but in this case camel case really seems to be more consistent and not required for "helm magic" to work.

File: image/templates/helm/stackrox-secured-cluster/values-scanner.yaml.example [hunk 13]
```diff
@@ -1,5 +1,5 @@
 # # NOTE:
-# # The Scanner is only available in the secured clusters on the OpenShift Container Platform.
+# # The Scanner is not available in the secured clusters installed via roxctl.
```
Comment (johannes94) [inline image/templates/helm/stackrox-secured-cluster/values-scanner.yaml.example:2]: Is this referring to scanner-slim(v2) being only available on OCP. Did that change?

I think what this comment should say is:
- scanner-slim is only available in the cluster on OCP
- scannerV4 is not available in secured clusters install via roxctl

File: operator/pkg/central/extensions/reconcile_tls.go [hunk 14]
```diff
@@ -74,6 +75,20 @@ func (r *createCentralTLSExtensionRun) Execute(ctx context.Context) error {
 	if err := r.ReconcileSecret(ctx, "scanner-db-tls", scannerEnabled && !shouldDelete, r.validateScannerDBTLSData, r.generateScannerDBTLSData, true); err != nil {
 		return errors.Wrap(err, "reconciling scanner-db-tls secret")
 	}
+
+	if features.ScannerV4.Enabled() {
```
Comment (johannes94) [inline operator/pkg/central/extensions/reconcile_tls.go:79]: Are operator changes in this PR strictly required to not break the operator? If not we should put them in another PR.

File: sensor/kubernetes/localscanner/service_certificates_repository.go [hunk 15]
```diff
@@ -69,6 +69,24 @@ func newServiceCertificatesRepo(ownerReference metav1.OwnerReference, namespace
 				serviceCertFileName: mtls.ServiceCertFileName,
 				serviceKeyFileName:  mtls.ServiceKeyFileName,
 			},
+			storage.ServiceType_SCANNER_V4_INDEXER_SERVICE: {
```
Comment (johannes94) [inline sensor/kubernetes/localscanner/service_certificates_repository.go:72]: Do I understand correctly that we're going to use the same certificate for all scanner components v2 and v4 combined?

File: central/credentialexpiry/service/service_impl.go [hunk 16]
```diff
@@ -46,6 +49,15 @@ func (s *serviceImpl) GetCertExpiry(ctx context.Context, request *v1.GetCertExpi
 		return s.getCentralCertExpiry()
 	case v1.GetCertExpiry_SCANNER:
 		return s.getScannerCertExpiry(ctx)
+	// TODO(ROX-20064): Needs to be implemented when Scanner V4 has been added to imageIntegrations.
```
Comment (johannes94) [inline central/credentialexpiry/service/service_impl.go:52]: I don't understands What exactly do you mean by "when Scanner V4 has been added to imageIntegrations"?

Isn't scannerV4 just another Integration of type ImageIntegration_ClairV4?

If so this should already be working or we only need to add the SN for scanner v4 to the matcher [here](https://github.com/stackrox/stackrox/blob/74e9d50a8fa8cf53bd3fe04438488828a1d0834f/central/credentialexpiry/service/service_impl.go#L103)

**Review by johannes94 — CHANGES_REQUESTED** [review]

File: image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl [hunk 17]
```diff
@@ -76,12 +76,12 @@
 {{- end -}}
 
 {{/* Configure images, certificates and passwords as required. */}}
-{{/* Note that for secured-cluster-services we don't configure certificates here, instead they will be distributed at runtime by Sensor and Central. */}}
+{{/* Note that for secured-cluster-services we usually don't configure certificates here, instead they will be distributed at runtime by Sensor and Central. */}}
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl:79]: You are right, this seems confusing. I think the reason why I added "usually" here is because for some of the standard tests contained in the helmtest test suite for secured-cluster-services the ability to inject certificates is quite important. I will add a note to the comment which hopefully clears it up.

File: image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/internal/config-shape.yaml:1]: Thanks for the clarification @johannes94. The answer is two-fold:

1. I had forgotten to insert the Scanner V4 structure for customization into central-services config-shape file. A follow-up PR for fixing this is in progress: https://github.com/stackrox/stackrox/pull/8996.

2. The remaining Scanner V4 related additions to the secured-cluster-services config-shape were required by secured-cluster's configuration structure: A top-level `image` field contains image-configurations for different components, e.g. `image.main`, `image.scanner`, etc. For consistency reasons I felt I have to add `image.scanner-v4-{indexer,matcher,db}` there. For central-services it looks different, because all Scanner V4 related configuration was contained below the top-level `scannerV4` field. And the config shape for *that* was pulled in from the shared folder.

File: sensor/kubernetes/localscanner/service_certificates_repository.go (hunk 15 above)
Comment (mclasmeier) [reply sensor/kubernetes/localscanner/service_certificates_repository.go:72]: No, different certificates for all scanner components.

File: operator/pkg/central/extensions/reconcile_tls.go (hunk 14 above)
Comment (mclasmeier) [reply operator/pkg/central/extensions/reconcile_tls.go:79]: I believe this was necessary to make the tests pass. I don't remember which test specifically, though.

File: central/credentialexpiry/service/service_impl.go (hunk 16 above)
Comment (dcaravel) [reply central/credentialexpiry/service/service_impl.go:52]: Thats right, ScannerV4 and ClairV4 are different.

Both use the `claircore` engine - but are different wrappers on top of the engine.  Features such as 'node scanning' will eventually be integrated into ScannerV4 but not ClairV4 (for example). We will be able to add ACS specific functionality into ScannerV4, something that we cannot do with ClairV4.

https://github.com/stackrox/stackrox/pull/8869 introduces the ScannerV4 integration (pending approval/review)

File: proto/api/v1/credential_expiry_service.proto [hunk 18]
```diff
@@ -12,9 +12,12 @@ package v1;
...
     enum Component {
-        UNKNOWN = 0;
-        CENTRAL = 1;
-        SCANNER = 2;
+        UNKNOWN            = 0;
+        CENTRAL            = 1;
+        SCANNER            = 2;
+        SCANNER_V4_INDEXER = 3;
+        SCANNER_V4_MATCHER = 4;
+        SCANNER_V4_DB      = 5;
```
Comment (dcaravel) [inline proto/api/v1/credential_expiry_service.proto:20]: From the service API perspective we probably just need the `SCANNER_V4` component (or bundle it all together with the existing `SCANNER` component if that makes sense).

Scanner + ScannerDB are bundled into one `Component`. In the UI there is only one action to take for `Scanner` today:

image

The `yaml` contains secrets for `scanner-tls` and `scanner-db-tls`, think it would makes sense that the cred expiry service will report on the 'soonest to expire' cert of `scanner-v4-indexer`, `scanner-v4-matcher`, and `scanner-v4-db`.    When the `yaml` is applied, it will refresh the certs for all the `scanner-v4` components.

File: sensor/kubernetes/clusterhealth/updater.go [hunk 19]
```diff
@@ -223,6 +223,8 @@ func (u *updaterImpl) getLocalScannerInfo() *storage.ScannerHealthInfo {
 		}
 	}
 
+	// TODO/MC: Extend for ScannerV4 case?
```
Comment (dcaravel) [inline sensor/kubernetes/clusterhealth/updater.go:226]: FYI, this is part of scope for ROX-21042 (and the what/how detailed in the ScannerV4 integration design)

**Review by dcaravel — COMMENTED** [review]
`features.ScannerV4.Enabled()` will be static and NOT indicate if ScannerV4 is installed.  

Tests for `features.ScannerV4.Enabled()` will likely  need to be replaced (or supplemented) by tests for the `ROX_SCANNER_V4_ENABLED` changeable feature being introduced in [ROX-21039](https://issues.redhat.com/browse/ROX-21039) (it's an [env var](https://github.com/stackrox/stackrox/blob/96e387547168fc1dd5c60d3f2cfa8e3bd958bec9/pkg/env/sensor.go#L27) currently but will be moved to a feature so that the front-end can benefit as needed) - in areas where running ScannerV4 components are expected (such as credential expiry and generation)

File: central/credentialexpiry/service/service_impl.go (hunk 16 above)
Comment (johannes94) [reply central/credentialexpiry/service/service_impl.go:52]: Thanks for explaining.

File: sensor/kubernetes/clusterhealth/updater.go (hunk 19 above)
Comment (mclasmeier) [reply sensor/kubernetes/clusterhealth/updater.go:226]: Good to know, thank you.

File: pkg/scannerv4/client/options.go [hunk 20]
```diff
@@ -5,17 +5,20 @@ import (
...
-	}
-
 	defaultOptions = options{
-		indexerOpts: defaultConnOptions,
-		matcherOpts: defaultConnOptions,
-		comboMode:   true,
+		indexerOpts: connOptions{
+			mTLSSubject: mtls.ScannerV4IndexerSubject,
+			address:     ":8443",
+			serverName:  "scanner-v4-indexer.stackrox", // TODO: allow custom namespaces
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply pkg/scannerv4/client/options.go:12]: Used that (minus the port), thank you.

File: pkg/scannerv4/client/options.go [hunk 21]
```diff
@@ -5,17 +5,20 @@ import (
...
+			skipTLS:     false,
+		},
+		matcherOpts: connOptions{
+			mTLSSubject: mtls.ScannerV4MatcherSubject,
+			address:     ":8443",
+			serverName:  "scanner-v4-matcher.stackrox",
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply pkg/scannerv4/client/options.go:18]: Same here (minus the port).

File: image/templates/helm/stackrox-secured-cluster/values-scanner-v4-indexer.yaml.example (hunk 12 above)
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/values-scanner-v4-indexer.yaml.example]: yes, this was simply wrong. In the code the kebabCase identifier were actually used (https://github.com/stackrox/stackrox/blob/d28a7cfcc5608335f4a01bae8506dc3a6233f7bb/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L295).

File: proto/api/v1/credential_expiry_service.proto (hunk 18 above)
Comment (mclasmeier) [reply proto/api/v1/credential_expiry_service.proto:20]: I have simplified this: now the only addition here is `SCANNER_V4`.

File: image/templates/helm/stackrox-secured-cluster/values-scanner.yaml.example (hunk 13 above)
Comment (mclasmeier) [reply image/templates/helm/stackrox-secured-cluster/values-scanner.yaml.example:2]: It seems this comes from commit c0cff8fcaf79ba52699144a6ed131380cd31bf5d from @RTann.
Ross, could you comment on this? I am pretty clueless here.

File: proto/storage/service_identity.proto [hunk 22]
```diff
@@ -19,19 +19,22 @@ message ServiceIdentity {
 
 // Next available tag: 13
 enum ServiceType {
```
Comment (johannes94) [inline proto/storage/service_identity.proto:21]: FYI: I've copied this changes to #8787 because they were required for TLS pre extension in the central reconciler.

Depending on timelines it might make sense to make those changes in an extra PR to get them in earlier.
